helmfile/charts/prometheus-alerts/templates/alerts/daily-checks.yaml

```
@@ -17,13 +17,36 @@ spec:
   groups:
   - name: daily-checks
     rules:
-    - alert: S3BucketContentOver{{.Values.s3BucketPercentLimit}}Percent
+{{- if .Values.s3BucketAlerts.size.enabled }}
+    - alert: S3BucketSizeOver{{.Values.s3BucketAlerts.size.percent}}Percent
       annotations:
-        description: Over {{.Values.s3BucketPercentLimit}} percent of the total s3bucket size limit is used.
-        summary: Over {{.Values.s3BucketPercentLimit}} percent of the total s3bucket size limit is used.
+        message: Over {{.Values.s3BucketAlerts.size.percent}} percent of the total s3bucket size limit is used in bucket {{`{{ $labels.bucket }}`}}.
       expr: |-
         (
-          sum(max_over_time(s3_objects_size_sum_bytes[1h])) >= ({{ .Values.s3BucketSizeQuotaGB }} * (1024^3)) * ({{ .Values.s3BucketPercentLimit }} / 100)
+          sum(s3_objects_size_sum_bytes{bucket!~"{{- range .Values.s3BucketAlerts.exclude }}{{ .  }}|{{- end }}"}) by (bucket) >= ({{ .Values.s3BucketAlerts.size.sizeQuotaGB }} * (1024^3)) * ({{ .Values.s3BucketAlerts.size.percent }} / 100)
+        )
+      for: 1h
+      labels:
+        severity: warning
+{{- end }}
+{{- if .Values.s3BucketAlerts.objects.enabled }}
+    - alert: S3BucketObjectsOver{{.Values.s3BucketAlerts.objects.percent}}Percent
+      annotations:
+        message: Over {{.Values.s3BucketAlerts.objects.percent}} percent of the total s3bucket object limit is used in bucket {{`{{ $labels.bucket }}`}}.
+      expr: |-
+        (
+          sum(s3_objects{bucket!~"{{- range .Values.s3BucketAlerts.exclude }}{{ .  }}|{{- end }}"}) by (bucket) > {{ .Values.s3BucketAlerts.objects.count }} * ({{ .Values.s3BucketAlerts.objects.percent }} / 100)
```

I just learned that there's a handy function for this, which also makes sure you don't get an extra `|` at the end
```suggestion
          sum(s3_objects{bucket!~"{{ join "|" .Values.s3BucketAlerts.exclude }}"}) by (bucket) > {{ .Values.s3BucketAlerts.objects.count }} * ({{ .Values.s3BucketAlerts.objects.percent }} / 100)
```

---

Awesome! :+1: